ecosystem/typescript/sdk/src/providers/aptos_client.ts

```diff
@@ -308,6 +316,91 @@ export class AptosClient {
     return builder.build(payload.function, payload.type_arguments, payload.arguments);
   }
 
+  /**
+   * Generates a fee payer transaction that can be signed and submitted to chain
+   *
+   * @param sender the sender's account address
+   * @param payload the transaction payload
+   * @param fee_payer the fee payer account
+   * @param secondary_signer_accounts an optional array of the secondary signers accounts
+   * @returns a fee payer raw transaction that can be signed and submitted to chain
+   */
+  async generateFeePayerTransaction(
+    sender: MaybeHexString,
+    payload: Gen.EntryFunctionPayload,
+    fee_payer: AptosAccount,
+    secondary_signer_accounts: Array<AptosAccount> = [],
+    options?: Partial<Gen.SubmitTransactionRequest>,
+  ): Promise<TxnBuilderTypes.FeePayerRawTransaction> {
+    const rawTxn = await this.generateTransaction(sender, payload, options);
+
+    const signers: Array<AccountAddress> = secondary_signer_accounts.map((account) =>
+      AccountAddress.fromHex(account.address()),
+    );
+
+    const feePayerTxn = new TxnBuilderTypes.FeePayerRawTransaction(
+      rawTxn,
+      signers,
+      AccountAddress.fromHex(fee_payer.address()),
+    );
+    return feePayerTxn;
+  }
+
+  /**
+   * Submits fee payer transaction to chain
+   *
+   * @param feePayerTransaction the raw transaction to be submitted, of type FeePayerRawTransaction
+   * @param senderAuthenticator the sender account authenticator (can get from signMultiTransaction() method)
+   * @param feePayerAuthenticator the feepayer account authenticator (can get from signMultiTransaction() method)
+   * @param signersAuthenticators an optional array of the signer account authenticators
+   * @returns The pending transaction
+   */
+  async submitFeePayerTransaction(
+    feePayerTransaction: TxnBuilderTypes.FeePayerRawTransaction,
+    senderAuthenticator: TxnBuilderTypes.AccountAuthenticatorEd25519,
+    feePayerAuthenticator: TxnBuilderTypes.AccountAuthenticatorEd25519,
+    signersAuthenticators: Array<TxnBuilderTypes.AccountAuthenticatorEd25519> = [],
```

Comment:
Let's drop `Authenticator` from the input names.

And we should name these:
1. sender
2. feePayer
3. additionalSigners

Additional because the sender and the fee payer are technically also signers

Reply:
I find adding `Authenticator` makes it clearer that we are expecting an `Authenticator` and not just an `account`

Reply:
okay, let's still rename the last one to additionalSignerAuthenticators